grep/cbeer/grep.go

```diff
@@ -0,0 +1,118 @@
+package grep
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+// Search a file for lines matching a regular expression pattern.
+func Search(pattern string, flags, files []string) []string {
+	searchResults := make([]string, 0)
+	var printLineNumbers bool
+	var printOnlyNames bool
+	var includeFileNames bool
+
+	for _, flagValue := range flags {
+		switch flagValue {
+		case "-n":
+			printLineNumbers = true
+		case "-l":
+			printOnlyNames = true
+		}
+	}
+
+	if len(files) > 1 {
+		includeFileNames = true
+	}
+
+	for _, fileName := range files {
+		fileHandle, err := os.Open(fileName)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer fileHandle.Close()
+
+		scanner := bufio.NewScanner(fileHandle)
+		lineNumber := 1
+		var lineOfText string
+
+		if printOnlyNames {
```

Comment:
Instead of having 2 separate loops for filename only and a "regular" one, you can have just one loop and break on the first match and include the filename in the results instead of lines.